File: javascript/easy/MergeTwoSortedLists_021.js
function ListNode(val) {
  this.val = val
  this.next = null
}

const merge = (l1, l2) => {
  if (l1 === null) {
    return l2
  }
  if (l2 === null) {
    return l1
  }
  if (l1.val > l2.val) {
    l2.next = merge(l1, l2.next)
    return l2
  } else {
    l1.next = merge(l1.next, l2)
    return l1
  }
}

console.log(merge(new ListNode(1), new ListNode(3)))

File: java/src/main/java/cn/az/code/easy/MergeTwoSortedLists_021.java
package cn.az.code.easy;

import cn.az.code.common.ListNode;

/**
 * @author : Liz
 * @date : 2019/10/26
 **/
public class MergeTwoSortedLists_021 {

    public static void main(String[] args) {

        ListNode l1 = new ListNode(1);
        l1.next = new ListNode(2);
        l1.next.next = new ListNode(5);
        ListNode l2 = new ListNode(1);
        l2.next = new ListNode(3);
        l2.next.next = new ListNode(4);
        System.out.println(merge(l1, l2));
    }

    public static ListNode merge(ListNode l1, ListNode l2) {
        if (l1 == null) {
            return l2;
        }
        if (l2 == null) {
            return l1;
        }
        if (l1.val > l2.val) {
            l2.next = merge(l1, l2.next);
            return l2;
        } else {
            l1.next = merge(l1.next, l2);
            return l1;
        }
    }
}
